### PR TITLE
ref(appconnect): Stronger typing and type fixes

### PR DESCRIPTION
**Pull request by flub**
This makes the appstore connect utilities entirely typed.
Also fixes up the types of some IDs which are numbers.

### CODE REVIEW

**Review by untitaker — APPROVED** [review]

**Review by Swatinem — APPROVED** [review]
lgtm, but I’m curious how stable the IDs are that we get from apple? will they ever change the ID from int to a uuid? might as well be a good idea to just treat it as an opaque blob

**Comment by flub** [comment]
> lgtm, but I’m curious how stable the IDs are that we get from apple? will they ever change the ID from int to a uuid? might as well be a good idea to just treat it as an opaque blob

hmm, the api seems to treat these as stable as you need to use them to query things.  I don't know if REST resource IDs are generally allowed to change for a resource?

as for the type... checking the api docs they do indeed say they're opaque ID with a string type.  so yes, i unified these in the wrong direction, i should have made them all strings rather than ints.  I'll update.

File: src/sentry/lang/native/symbolicator.py [hunk 1]
```diff
@@ -61,10 +61,11 @@
         "itunesUser": {"type": "string", "minLength": 1, "maxLength": 100},
         "itunesCreated": {"type": "string", "format": "date-time"},
         "itunesPassword": {"type": "string"},
-        "itunesSession": {"type": "string"},
         "itunesPersonId": {"type": "string"},
+        "itunesSession": {"type": "string"},
         "appName": {"type": "string", "minLength": 1, "maxLength": 512},
-        "appId": {"type": "string", "minLength": 1, "maxLength": 512},
+        "appId": {"type": "string", "minLength": 1},
+        "bundleId": {"type": "string", "minLength": 1},
```
Comment (priscilawebdev) [inline src/sentry/lang/native/symbolicator.py:68]: should we also add this property to the required list?

Comment (flub) [reply src/sentry/lang/native/symbolicator.py:68]: good catch.  done

**Review by priscilawebdev — APPROVED** [review]
Tested everything locally and it looks good to me :)

**Pull request merged**